src/main/java/ch/njol/skript/expressions/ExprQueue.java

```diff
@@ -0,0 +1,96 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Feature;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.util.SkriptQueue;
+
+import java.util.Iterator;
+
+@Name("Queue (Experimental)")
+@Description("""
+	Requires the <code>using queues</code> experimental feature flag to be enabled.
+	
+	Creates a new queue.
+	A queue is a set of elements that can have things removed from the start and added to the end.
+	
+	Any value can be added to a queue. Adding a non-existent value (e.g. `{variable that isn't set}`) will have no effect.
+	This means that removing an element from the queue will always return a value <i>unless the queue is empty</i>.
+	
+	Requesting an element from a queue (e.g. `the 1st element of {queue}`) also removes it from the queue.""")
+@Examples({
+	"""
+		set {queue} to a new queue
+		add "hello" and "there" to {queue}
+		broadcast the first element of {queue} # hello
+		broadcast the first element of {queue} # there
+		# queue is now empty""",
+	"""
+		set {queue} to a new queue of "hello" and "there"
+		broadcast the last element of {queue} # removes 'there'
+		add "world" to queue
+		broadcast the first 2 elements of {queue} # removes 'hello', 'world'
+		"""
```

Comment:
Everything's indented twice except the starting """. Why isn't it all on the same level?

Reply:
That's how it's supposed to be, I believe,
the expression starts from `"""` but then there's a mandatory new-line
and our conventions say you have to indent stuff when you new-line.

It only looks weird because there's nothing _before_ the start as you'd normally have, e.g. `String ... = """` like this
<img width="225" alt="image" src="https://github.com/user-attachments/assets/c1953be7-e494-40b3-9e3d-76924a7114fc">